test/client/js/event-target.js

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright 2015-2016, Google, Inc.
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* global describe it */
+'use strict';
+import EventTarget from '../../../public/js/event-target';
+const assert = require('assert');
+
+describe('js.event-target', () => {
+  describe('#addEventLister', () => {
+    it('Add a single listener to a type', () => {
+      const eventTarget = new EventTarget();
+      const callback = () => {};
+      eventTarget.addEventListener('type', callback);
+      const eventListeners = eventTarget.getEventListeners('type');
+      assert.notEqual(eventListeners, null);
+      assert.equal(eventListeners.size, 1);
+      assert.equal(eventListeners.has(callback), true);
```

Comment:
I wouldn't test whether an callback exists (or whether event listeners has size 1. These are implementation details. Testing implementation details make tests fragile. Instead focus tests on the behavior that is visible from the outside -> event dispatching. I